action.yml

```diff
@@ -0,0 +1,79 @@
+# action.yml
+name: 'lazy-action-dotnet'
+description: 'Github action to perform build, test , scan and generate image.'
+inputs:
+  src_file_dir:  # source file directory
+    description: 'Source file directory'
+    required: true
+    default: './'
+  dockerfile_dir:  # docker file directory
+    description: 'Dockerfile  directory'
+    required: true
+  github_token:  # github owner token
+    description: 'Github token'
+    required: true
+  github_owner_token:  # github owner token
+    description: 'Github owner token'
+    required: true
+  sonar_token:  # sonar token
+    description: 'Github token'
+    required: true
+  dev_aws_account_id:  # dev aws account id
+    description: 'Dev AWS Account ID'
+    required: true 
+  aws_access_key_id:  # aws access key id
+    description: 'AWS access key'
+    required: true
+  aws_secret_access_key:  # aws secret access key
+    description: 'AWS secret access key'
+    required: true
+  aws_region:  # aws region
+    description: 'AWS Region'
+    required: false
+    default: 'us-east-2'
+  docker_repo_name:  # docker repo name 
+    description: 'Docker repo name'
+    required: true
+  docker_image_name:  # docker image name 
+    description: 'Docker image name'
+    required: true
+  sonar_project_key:  #  sonar project key
+    description: 'Sonar Project Key'
+    required: false
+    default: 'variant-inc'
+  sonar_org:  # sonar organization
+    description: 'Sonar organization'
+    required: false
+    default: 'ariant'
+  sonar_scan_enabled:  # sonar scan enabled
+    description: 'To check sonar scan needed.'
+    required: true
+  nuget_push:  # build publish
+    description: 'build publish'
+    required: true
+  docker_push:  # docker push
+    description: 'docker push'
+    required: true
+
+
+runs:
+  using: 'docker'
+  image: docker://public.ecr.aws/p4v7w0a5/lazy/donet-base
+  env:
```

Comment:
I don't think you need to send env like this. Even if you don't provide the input will come as `INPUT_<variable>`

Reply:
Changed to Input_ variables